### PR TITLE
Set pad_token in run_glue_no_trainer.py #28534

### LINKED ISSUE #28534: run_glue_no_trainer.py script crashes on Mistral model due to tokenizer issue
### System Info

- `transformers` version: 4.36.2
- Platform: Linux-5.15.0-91-generic-x86_64-with-glibc2.35
- Python version: 3.10.13
- Huggingface_hub version: 0.19.4
- Safetensors version: 0.4.0
- Accelerate version: 0.25.0
- Accelerate config:    - compute_environment: LOCAL_MACHINE
        - distributed_type: DEEPSPEED
        - mixed_precision: bf16
        - use_cpu: False
        - debug: False
        - num_processes: 8
        - machine_rank: 0
        - num_machines: 1
        - rdzv_backend: static
        - same_network: True
        - main_training_function: main
        - deepspeed_config: {'gradient_accumulation_steps': 1, 'offload_optimizer_device': 'none', 'offload_param_device': 'none', 'zero3_init_flag': True, 'zero3_save_16bit_model': False, 'zero_stage': 3}
        - downcast_bf16: no
        - tpu_use_cluster: False
        - tpu_use_sudo: False
        - tpu_env: []
- PyTorch version (GPU?): 2.1.1+cu121 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): 0.7.5 (cpu)
- Jax version: 0.4.21
- JaxLib version: 0.4.21
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: Yes

### Who can help?

@ArthurZucker @younesbelkada @pacman100 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Check out the transformers repo, and run this command (on a large server with appropriately configured `accelerate`, so it won't OOM):

`python run_glue_no_trainer.py   --model_name_or_path mistralai/Mistral-7B-v0.1   --task_name sst2   --per_device_train_batch_size 4   --learning_rate 2e-5   --num_train_epochs 3   --output_dir /tmp/sst2`

It will crash with this error and stack trace:

```
You're using a LlamaTokenizerFast tokenizer. Please note that with a fast tokenizer, using the `__call__` method is faster than using a method to encode the text followed by a call to the `pad` method to get a padded encoding.
Traceback (most recent call last):
  File "/scratch/brr/run_glue.py", line 662, in <module>
    main()
  File "/scratch/brr/run_glue.py", line 545, in main
    for step, batch in enumerate(active_dataloader):
  File "/scratch/miniconda3/envs/brr/lib/python3.10/site-packages/accelerate/data_loader.py", line 448, in __iter__
    current_batch = next(dataloader_iter)
  File "/scratch/miniconda3/envs/brr/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 630, in __next__
    data = self._next_data()
  File "/scratch/miniconda3/envs/brr/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 674, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/scratch/miniconda3/envs/brr/lib/python3.10/site-packages/torch/utils/data/_utils/fetch.py", line 54, in fetch
    return self.collate_fn(data)
  File "/scratch/miniconda3/envs/brr/lib/python3.10/site-packages/transformers/data/data_collator.py", line 249, in __call__
    batch = self.tokenizer.pad(
  File "/scratch/miniconda3/envs/brr/lib/python3.10/site-packages/transformers/tokenization_utils_base.py", line 3259, in pad
    padding_strategy, _, max_length, _ = self._get_padding_truncation_strategies(
  File "/scratch/miniconda3/envs/brr/lib/python3.10/site-packages/transformers/tokenization_utils_base.py", line 2707, in _get_padding_truncation_strategies
    raise ValueError(
ValueError: Asking to pad but the tokenizer does not have a padding token. Please select a token to use as `pad_token` `(tokenizer.pad_token = tokenizer.eos_token e.g.)` or add a new pad token via `tokenizer.add_special_tokens({'pad_token': '[PAD]'})`.
/scratch/miniconda3/envs/brr/lib/python3.10/tempfile.py:860: ResourceWarning: Implicitly cleaning up <TemporaryDirectory '/tmp/tmprbynkmzk'>
  _warnings.warn(warn_message, ResourceWarning)
```

### Expected behavior

It should train without crashing.

### PR DESCRIPTION
**Pull request by JINO-ROHIT**
# What does this PR do?

This PR adds the pad tokenizer in the run_glue_no_trainer.py script under examples.

Fixes #28534 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@amyeroberts @ArthurZucker

### CODE REVIEW

File: examples/pytorch/text-classification/run_glue_no_trainer.py [hunk 1]
```diff
@@ -327,6 +327,9 @@ def main():
     tokenizer = AutoTokenizer.from_pretrained(
         args.model_name_or_path, use_fast=not args.use_slow_tokenizer, trust_remote_code=args.trust_remote_code
     )
+    tokenizer.pad_token = tokenizer.eos_token
+    config.pad_token_id = tokenizer.pad_token_id
```
Comment (amyeroberts) [inline examples/pytorch/text-classification/run_glue_no_trainer.py:331]: We shouldn't override the pad_token if it's already set

```suggestion
    if tokenizer.pad_token is None: 
        tokenizer.pad_token = tokenizer.eos_token
    config.pad_token_id = tokenizer.pad_token_id
```

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding - just a small suggestion

File: examples/pytorch/text-classification/run_glue_no_trainer.py (hunk 1 above)
Comment (JINO-ROHIT) [reply examples/pytorch/text-classification/run_glue_no_trainer.py:331]: that makes sense, sure ill update it

**Comment by JINO-ROHIT** [comment]
done! @amyeroberts

**Review by amyeroberts — APPROVED** [review]
Thanks for iterating! 

For the failing tests, running `make fixup` and pushing the changes should resolve

**Comment by JINO-ROHIT** [comment]
Exception: Found the following copy inconsistencies:
- tests/models\roc_bert\test_tokenization_roc_bert.py: copy does not match models.bert.test_tokenization_bert.BertTokenizationTest.test_is_whitespace at line 167
Run `make fix-copies` or `python utils/check_copies.py --fix_and_overwrite` to fix them.
make: *** [Makefile:38: repo-consistency] Error 1

I get this error but i tried running both commands, still gives the same error, how to resolve this?

**Comment by amyeroberts** [comment]
There's something funny going on with the formatting here - files like src/transformers/models/bigbird_pegasus/modeling_bigbird_pegasus.py shouldn't be modified when calling `make fixup`.
and you shouldn't need to run `make fix-copies` for this PR. 

In your environment, make sure you have all of the necessary formatting libraries with `pip install -e .[testing]`. Once that's done, I'd undo the formatting changes to the unrelated files and then try running `make fix up` again

**Comment by JINO-ROHIT** [comment]
@amyeroberts ive recreated my env, can you help me out with undoing the formatting files?

**Pull request closed without merging**